proofs_test.go

```diff
@@ -3,9 +3,168 @@ package smt
 import (
```

Comment:
Optional: can be done in a future ommit.

Adding a .json text fixture to:
1. show what it looks like serialized as json
2. Test loading from a .json file.

Reply:
My thinking around using JSON was not that it would be loaded from JSON files but instead just an easy means to serialise, I would use `gob` but I opened issue #22 to standardise this into proto definitions probably.

I don't see a benefit to adding the above when the methods could do any encoding under the hood. The serialised `[]byte` is what would ideally be passed around.

LMK if you think differently

Reply:
If we only need to pass around `[]byte` (accomplished by gob) and eventually you into `proto`, why are we introducing the interim `json` if `gob` gets the job done until we move to `proto`?

Reply:
@Olshansk fixed uses gob now